designs/2022-lintFiles-lintText-fix-support/README.md

```diff
@@ -0,0 +1,251 @@
+- Repo: eslint/eslint
+- Start Date: 2022-05-06
+- RFC PR: 
+- Authors: SherrybabyOne
+
+# add fix support of eslint.lintFiles, eslint.lintText
+
+## Summary
+
+[eslint.lintFiles](https://eslint.org/docs/developer-guide/nodejs-api#-eslintlintfilespatterns) and [eslint.lintText](https://eslint.org/docs/developer-guide/nodejs-api#-eslintlinttextcode-options) add parameter fix to control rule checking.
+
+## Motivation
+
+<!-- Why are we doing this? What use cases does it support? What is the expected
+outcome? -->
+
+I want  `eslint.lintText()` and `eslint.lintFiles()` provide the **fix** parameter under options to control whether to fix the rules or not.
+
+This way you can use the same eslint instance for both fix and no-fix rule checking instead of having to create two ESLint instances:
+
+```javascript
+const eslint = new ESLint();
+
+// no fix
+const resultsForDiagnose = await eslint.lintText(text, { filePath, fix: false });
+// fix
+const resultsForFormat = await eslint.lintText(text, { filePath, fix: true });
+
+// no fix
+const lintFilesForDiagnose = await eslint.lintFiles(files, { fix: false });
+// fix
+const lintFilesForFormat = await eslint.lintFiles(files, { fix: true });
+```
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+### eslint.lintText(code, options)
+
+The options of `lintText(code, options)` support the fix parameter and pass it to `cliEngine.executeOnText()`:
```

Comment:
If the change being accepted, how is the `new ESLint({fix: true})` working?

Reply:
If `options.fix` in `eslint.lintText() `is **undefined,** then `options.fix` in `ESLint.constructor` will be followed

In other words, fix in `lintText` and `lintFiles` have higher priority than fix in `ESLint.constructor`

Reply:
so it won't be a breaking change. makes sense to me! 👍

Reply:
and I'd suggest soft-deprecating `opts.fix` in `ESLint()`, and remove it in future major versions - it's always no good to have two ways to do the same thing. :)

Reply:
@SherrybabyOne please add your answer into the RFC